### PR TITLE
Fix <base> tag rendering.

### PR DESCRIPTION
**Pull request by eskimor**
The generated uris for `static @` should really be absolute and not relative. In a project of mine, we now have some frontend uris which change the path to no longer being "/", without this change static resource loading is broken.

### CODE REVIEW

**Comment by 3noch** [comment]
Ideally the `/static` portion would be determined by some "root static path" configuration/route instead of hardcoded.

**Comment by ryantrinkle** [comment]
@eskimor The way we do this right now is to add a `<base>` tag to the head, which should always set the app's base path to the same place.  If this isn't working, we should see if we can fix it some other way.

The issue with making `static` produce an absolute path is that then that path would have to be baked in at compile-time, would have to change between platforms, etc.  Seems complicated, to me.

Can you give some more info on the problems you're seeing?

**Comment by 3noch** [comment]
Should obelisk have a way to set up `<base>` for you based on route config?

**Comment by ryantrinkle** [comment]
@3noch It does: see the head of this: view-source:https://reflex-frp.org/docs

**Comment by ryantrinkle** [comment]
Oh, maybe it's not based on the route config; but it definitely sets the base for you.

**Comment by 3noch** [comment]
Oh this must be a newer feature. Awesome.

**Comment by ryantrinkle** [comment]
This was added along with routes, because otherwise routes are constantly broken in the exact way @eskimor described.

**Comment by eskimor** [comment]
Thanks @ryantrinkle  - I was not aware of the \<base\> tag. What was causing problems for pact-web was twofold:

1. The base tag does not get added in `renderJsaddleFrontend`.
2. It gets added in `renderGhcjsFrontend` - but only after the head of the frontend gets rendered, which is too late if the head of the frontend already contained some references (which is the case for pact-web).

Both are now fixed by this PR.

File: lib/backend/src/Obelisk/Backend.hs [hunk 1]
```diff
@@ -171,5 +171,5 @@ renderGhcjsFrontend urlEnc route f = do
       ghcjsPreload = elAttr "link" ("rel" =: "preload" <> "as" =: "script" <> "href" =: "ghcjs/all.js") blank
       ghcjsScript = elAttr "script" ("language" =: "javascript" <> "src" =: "ghcjs/all.js" <> "defer" =: "defer") blank
   liftIO $ renderFrontendHtml urlEnc route
-    (_frontend_head f >> injectExecutableConfigs >> baseTag >> ghcjsPreload)
+    (baseTag >> _frontend_head f >> injectExecutableConfigs >> ghcjsPreload)
```
Comment (ryantrinkle) [inline lib/backend/src/Obelisk/Backend.hs]: Huh, I wouldn't have expected the order to matter here; can you elaborate, and maybe add a comment mentioning that it's important?

File: lib/run/src/Obelisk/Run.hs [hunk 2]
```diff
@@ -152,7 +152,8 @@ renderJsaddleFrontend :: (route -> Text) -> route -> Frontend route -> IO ByteSt
 renderJsaddleFrontend urlEnc r f =
   let jsaddleScript = elAttr "script" ("src" =: "/jsaddle/jsaddle.js") blank
       jsaddlePreload = elAttr "link" ("rel" =: "preload" <> "as" =: "script" <> "href" =: "/jsaddle/jsaddle.js") blank
-  in renderFrontendHtml urlEnc r (_frontend_head f >> injectExecutableConfigs >> jsaddlePreload) (_frontend_body f >> jsaddleScript)
+      baseTag  = elAttr "base" ("href" =: "/") blank --TODO: Figure out the base URL from the routes
+  in renderFrontendHtml urlEnc r (baseTag >> _frontend_head f >> injectExecutableConfigs >> jsaddlePreload) (_frontend_body f >> jsaddleScript)
```
Comment (ryantrinkle) [inline lib/run/src/Obelisk/Run.hs]: Is it possible to share the rendering logic with Obelisk.Backend?

Comment (eskimor) [reply lib/run/src/Obelisk/Run.hs]: I'll look into it!

File: lib/backend/src/Obelisk/Backend.hs (hunk 1 above)
Comment (eskimor) [reply lib/backend/src/Obelisk/Backend.hs]: Will do and I will test again to make really sure - but I tested twice already, so I am pretty sure already.

**Comment by eskimor** [comment]
@ryantrinkle ok moved the common parts to `renderFrontendHtml` where they should be - at least that what makes sense to me and added comment about the order.

File: lib/run/src/Obelisk/Run.hs (hunk 2 above)
Comment (eskimor) [reply lib/run/src/Obelisk/Run.hs]: Done

**Comment by ryantrinkle** [comment]
Thanks!

**Pull request merged**